src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs

```diff
@@ -105,6 +114,20 @@ public override string SubProtocol
                 websocketControl.SupportedProtocols.Add(subProtocol);
             }
 
+            if (MessageWebSocketClientCertificateSupported)
+            {
+                // Use the first app-provided client cert that can be successfully converted into a WinRT client cert (if any).
+                foreach (X509Certificate dotNetClientCert in options.ClientCertificates)
+                {
+                    RTCertificate winRtClientCert = ConvertDotNetClientCertToWinRtClientCert(dotNetClientCert);
+                    if (winRtClientCert != null)
```

Comment:
In the new iteration, I'm throwing a NotSupportedException if an app has specified some client certs and first cert cannot be converted into a WinRT Certificate. The exception is being thrown as part of _WinRTWebSocket.ConnectAsync_ following the same rationale explained in the other comment thread.